include/circt/Dialect/ESI/ESIPorts.td

```diff
@@ -67,3 +67,45 @@ def UnwrapValidReady : ESI_Op<"unwrap.vr", [NoSideEffect]> {
   let printer = [{ return ::print(p, *this); }];
   let parser = [{ return ::parse$cppClass(parser, result); }];
 }
+
+def ModportType:
+  Type<CPred<"$_self.isa<::circt::sv::ModportType>()">, "sv.interface">;
+
+def WrapSVInterface: ESI_Op<"wrap.iface", [NoSideEffect]> {
+  let summary = "Wrap an SV interface into an ESI port";
+  let description = [{
+    Wrap a SystemVerilog interface into an ESI channel. Interface MUST look
+    like an interface produced by ESI meaning it MUST contain valid, ready,
+    and data signals. Any other signals will be discarded.
+  }];
+
+  let arguments = (ins ModportType:$iface);
+  let results = (outs ChannelType:$output);
+
+  let assemblyFormat = [{
+    $iface attr-dict `:` type($iface) `->` type($output)
+  }];
+
+  let verifier = [{
+    return ::verify$cppClass(*this);
+  }];
+}
+
+def UnwrapSVInterface : ESI_Op<"unwrap.iface", [NoSideEffect]> {
+  let summary = "Unwrap an SV interface from an ESI port";
+  let description = [{
+    Unwrap an ESI channel into a SystemVerilog interface containing valid,
+    ready, and data signals.
+  }];
+
+  let arguments = (ins ChannelType:$chanInput, ModportType:$outIface);
```

Comment:
Why make `outIface` an "out parameter" here instead of just returning it as a result? I guess it depends how you wanted to use it, but I imagined this op would take a channel and return an SSA value of type `sv::ModportType`.

Reply:
That makes too much sense...  I discovered this the hard way after writing a bunch of code assuming that this made sense: returning a `Modport` value is worthless since you can't assign it to an `Interface`!

It also doesn't make sense to add an op to assign a modport to an interface since you can't do that in SystemVerilog. I'm now of the opinion that interfaces don't make a lot of sense in an SSA world and have come to the conclusion that we need an `sv.module` with all the ports as arguments and no results. But that's an argument for another time.

Reply:
Got it, that makes sense.

Reply:
About the hypothetical `sv.module` that has all it's ports as arguments... I guess I was thinking we could use `rtl.module` with all it's ports as SV interface arguments to achieve this for now.

Reply:
I would say abuse rather than use, but I was thinking the same.